tapir/shifts/views/management.py

```diff
@@ -90,29 +95,47 @@ class CancelShiftView(LoginRequiredMixin, PermissionRequiredMixin, UpdateView):
     template_name = "shifts/cancel_shift.html"
 
     def form_valid(self, form):
-        with transaction.atomic():
-            response = super().form_valid(form)
+        response = super().form_valid(form)
+        shift: Shift = form.instance
+        ShiftCancellationService.cancel(shift)
+        return response
 
-            shift: Shift = form.instance
-            shift.cancelled = True
-            shift.save()
-
-            for slot in shift.slots.all():
-                attendance = slot.get_valid_attendance()
-                if not attendance:
-                    continue
-                if (
-                    hasattr(slot.slot_template, "attendance_template")
-                    and slot.slot_template.attendance_template.user == attendance.user
-                ):
-                    attendance.state = ShiftAttendance.State.MISSED_EXCUSED
-                    attendance.excused_reason = "Shift cancelled"
-                    attendance.save()
-                    attendance.update_shift_account_entry()
-                else:
-                    attendance.state = ShiftAttendance.State.CANCELLED
-                    attendance.save()
 
+class CancelDayShiftsView(LoginRequiredMixin, PermissionRequiredMixin, FormView):
+    permission_required = PERMISSION_SHIFTS_MANAGE
+    form_class = BulkShiftCancelForm
+    template_name = "shifts/shift_day_cancel.html"
+    success_url = "/shifts/calendar"
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        day = datetime.datetime.strptime(self.kwargs["day"], "%d-%m-%y").date()
+        kwargs["shifts"] = Shift.objects.filter(start_time__date=day).order_by(
+            "start_time"
+        )
+        return kwargs
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data()
+        day = datetime.datetime.strptime(self.kwargs["day"], "%d-%m-%y").date()
+        context["day"] = day
+
+        return context
+
+    def form_valid(self, form):
+        with transaction.atomic():
+            response = super().form_valid(form)
+            cancellation_reason = form.cleaned_data["cancellation_reason"]
+            shift_ids_to_cancel = [
+                k.split("_", maxsplit=1)[-1]
+                for k, v in form.cleaned_data.items()
```

Comment:
Let's use more descriptive names for `k` and `v`, maybe `form_field_key` and `should_be_cancelled`